sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -1774,6 +1774,45 @@ class CastSuite extends CastSuiteBase {
       assert(e3.contains("Casting 2147483648 to int causes overflow"))
     }
   }
+
+  test("SPARK-35111: Cast string to year-month interval") {
+    checkEvaluation(cast(Literal.create("INTERVAL '0-0' YEAR TO MONTH"),
```

Comment:
This is covered by the `Seq("INTERVAL '0-0' YEAR TO MONTH" ...` below.

Reply:
Done